src/library-aux/scala/Any.scala

```diff
@@ -138,13 +138,87 @@ abstract class Any {
    */
   final def isInstanceOf[T0]: Boolean = sys.error("isInstanceOf")
 
-  /** Cast the receiver object to be of type `T0`.
-   *
-   *  Note that the success of a cast at runtime is modulo Scala's erasure semantics.
-   *  Therefore the expression `1.asInstanceOf[String]` will throw a `ClassCastException` at
-   *  runtime, while the expression `List(1).asInstanceOf[List[String]]` will not.
-   *  In the latter example, because the type argument is erased as part of compilation it is
-   *  not possible to check whether the contents of the list are of the requested type.
+  /** Tell the compiler to skip type safety and treat the receiver object to be of type `T0`.
+   *
+   *  This method is intended for cases where the compiler isn't able to know or capable to infer `T0` and neither
+   *  is it possible to check the subtype relation at runtime and hence skipping type safety is the only option.
+   *
+   *  DON'T USE `asInstanceOf` for type checking, it is **not** an alias for something like
```

Comment:
```suggestion
   *  DO NOT USE `asInstanceOf` for type checking, it is **not** an alias for something like
```

Nitpick: Do not use short forms when stressing your point !

Reply:
good point about stressing the point :D